library/ecp_curves.c

```diff
@@ -5213,40 +5214,29 @@ static int ecp_mod_p521( mbedtls_mpi *N )
 
 /*
  * Fast quasi-reduction modulo p255 = 2^255 - 19
- * Write N as A0 + 2^255 A1, return A0 + 19 * A1
+ * Write N as A0 + 2^256 A1, return A0 + 38 * A1
  */
 static int ecp_mod_p255( mbedtls_mpi *N )
 {
-    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-    size_t i;
-    mbedtls_mpi M;
-    mbedtls_mpi_uint Mp[P255_WIDTH + 2];
+    mbedtls_mpi_uint Mp[P255_WIDTH];
 
-    if( N->n < P255_WIDTH )
+    /* Helper references for top part of N */
+    mbedtls_mpi_uint * const NT_p = N->p + P255_WIDTH;
+    const size_t NT_n = N->n - P255_WIDTH;
+    if( NT_n == 0 || NT_n > P255_WIDTH )
```

Comment:
sorry, I'm not understanding this. the original test was `if( N->n < P255_WIDTH ) return 0;`. do you just mean you wanted to change this to `if( N->n <= P255_WIDTH ) return 0;` (and then calculate `NT_n` afterwards)? because that's probably simpler than putting `if( NT_n == 0 || NT_n > N->n ) return 0;` at line 5226 (and since `NT_n` is `N->n - P255_WIDTH` I don't see that it's the same, anyway - what am I missing?

Reply:
You're right, one should stick with the original check. I preferred the old one because it didn't use `N->n` anymore, but well, it was wrong, and fixing it reintroduces `N->n` anyway.

Reply:
ah, got it now, you're doing an unsigned wrap-around check. why not just do `if( N->n <= P255_WIDTH ) return 0;` earlier on? might be more obvious, and only one comparison